File: mephisto/client/review_app/server/urls.py
from flask import Flask

from .api import views as api_views


def init_urls(app: Flask):
    app.add_url_rule(
        "/qualifications/<int:qualification_id>/workers",
        view_func=api_views.QualificationWorkersView.as_view("qualification_workers"),
    )
    app.add_url_rule(
        "/qualifications/<int:qualification_id>/workers/<int:worker_id>/grant",
        view_func=api_views.QualifyWorkerView.as_view("qualification_worker_grant"),
        defaults={"action": "grant"},
    )
    app.add_url_rule(
        "/qualifications/<int:qualification_id>/workers/<int:worker_id>/revoke",
        view_func=api_views.QualifyWorkerView.as_view("qualification_worker_revoke"),
        defaults={"action": "revoke"},
    )
    app.add_url_rule(
        "/qualifications/",
        view_func=api_views.QualificationsView.as_view("qualifications"),
    )
    app.add_url_rule(
        "/tasks/",
        view_func=api_views.TasksView.as_view("tasks"),
    )
    app.add_url_rule(
        "/tasks/<int:task_id>/worker-units-ids",
        view_func=api_views.TasksWorkerUnitsView.as_view("worker-units-ids"),
    )
    app.add_url_rule(
        "/units",
        view_func=api_views.UnitsView.as_view("units"),
    )
    app.add_url_rule(
        "/units/details",
        view_func=api_views.UnitsDetailsView.as_view("units_details"),
    )
    app.add_url_rule(
        "/units/approve",
        view_func=api_views.UnitsApproveView.as_view("units_approve"),
    )
    app.add_url_rule(
        "/units/reject",
        view_func=api_views.UnitsRejectView.as_view("units_reject"),
    )
    app.add_url_rule(
        "/units/soft-reject",
        view_func=api_views.UnitsSoftRejectView.as_view("units_soft_reject"),
    )
    app.add_url_rule(
        "/workers/<int:worker_id>/block",
        view_func=api_views.WorkerBlockView.as_view("worker_block"),
    )
    app.add_url_rule(
        "/workers/<int:worker_id>/stats",
        view_func=api_views.WorkerStatsView.as_view("worker_stats"),
    )

File: mephisto/client/review_app/client/src/urls.ts
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

const API_URL = "http://localhost:5001";

const urls = {
  client: {
    home: "/",
    task: (id) => `/tasks/${id}`,
    tasks: "/tasks",
  },
  server: {
    qualifications: API_URL + "/qualifications",
    qualificationWorkers: (id) => API_URL + `/qualifications/${id}/workers`,
    qualificationGrantWorker: (id, workerId) =>
      API_URL + `/qualifications/${id}/workers/${workerId}/grant`,
    qualificationRevokeWorker: (id, workerId) =>
      API_URL + `/qualifications/${id}/workers/${workerId}/revoke`,
    stats: API_URL + "/stats",
    task: (id) => API_URL + `/tasks/${id}`,
    tasks: API_URL + "/tasks",
    tasksWorkerUnitsIds: (id) => API_URL + `/tasks/${id}/worker-units-ids`,
    unitReviewHtml: (id) => API_URL + `/units/${id}/review.html`,
    units: API_URL + "/units",
    unitsApprove: API_URL + "/units/approve",
    unitsDetails: API_URL + "/units/details",
    unitsReject: API_URL + "/units/reject",
    unitsSoftReject: API_URL + "/units/soft-reject",
    workersBlock: (id) => API_URL + `/workers/${id}/block`,
  },
};

export default urls;
